src/server/message.rs

```diff
@@ -359,21 +388,57 @@ mod test {
 
     // http://www.jsonrpc.org/specification#request_object
     #[test]
-    fn parse_raw_message() {
-        let raw_msg = json!({
+    fn raw_message_parses_valid_jsonrpc_request_with_string_id() {
+        let raw_json = json!({
+            "jsonrpc": "2.0",
+            "id": "abc",
+            "method": "someRpcCall",
+        }).to_string();
+
+        let expected_msg = RawMessage {
+            method: "someRpcCall".to_owned(),
+            id: Id::Str("abc".to_owned()),
+            // Internally missing parameters are represented as null
+            params: serde_json::Value::Null,
+        };
+        assert_eq!(expected_msg, RawMessage::try_parse(&raw_json).unwrap().unwrap());
+    }
+
+    #[test]
+    fn raw_message_parses_valid_jsonrpc_request_with_numeric_id() {
+        let raw_json = json!({
             "jsonrpc": "2.0",
             "id": "1",
             "method": "someRpcCall",
         }).to_string();
 
-        let str_msg = RawMessage {
+        let expected_msg = RawMessage {
             method: "someRpcCall".to_owned(),
-            // FIXME: for now we support only numeric ids
-            id: Some(Id::Num(1)),
+            id: Id::Num(1),
             // Internally missing parameters are represented as null
             params: serde_json::Value::Null,
         };
-        assert_eq!(str_msg, RawMessage::try_parse(&raw_msg).unwrap().unwrap());
+        assert_eq!(expected_msg, RawMessage::try_parse(&raw_json).unwrap().unwrap());
+    }
+
+    #[test]
+    fn raw_message_with_string_id_parses_into_request() {
+        #[derive(Debug)]
+        pub enum DummyRequest { }
+        impl LSPRequest for DummyRequest {
+            type Params = ();
+            type Result = ();
+            const METHOD: &'static str = "dummyRequest";
+        }
+
+        let raw_msg = RawMessage {
+            method: "dummyRequest".to_owned(),
+            id: Id::Str("abc".to_owned()),
+            params: serde_json::Value::Null,
+        };
+
+        let request: Request<DummyRequest> = raw_msg.parse_as_request().expect("RawMessage with numeric id should parse into request");
```

Comment:
I think the error message may be wrong? We parse RawMessage with a string id, not a numeric one

also minor nit: I think it'd be good to split this long line, running `rustfmt` on the code and checking in appropriate snippet should do the trick :)

Reply:
Good catch, fixed. I also run rustfmt, but it applied more changes than just my changed lines. I believe it is because this file was not reformatted after rustfmt version bump. But if that is not a case, and I should only format the lines I changed, let me know and I will update the PR.